### PR TITLE
dev: setup-pyenv and bake in PYENV_VERSION into setup-git

### PR DESCRIPTION
**Pull request by joshuarli**
When setting up pre-commit via `make setup-git` for the first time, in certain cases, black installation will fail because pyenv fails to find python3:

	pyenv: python3: command not found

This is due to `.python-version` only having 2.7, causing pyenv to mask the system or brew python3 (again, in some cases and configurations, lol).

In the past I've prescibed `PYENV_VERSION=3.whatever make setup-git` as a one-time workaround, but this issue has cropped up more times than I'd like.

If multiple versions are present in `.python-version`, pyenv will use them.
Unfortunately, `pyenv install` by itself cannot parse multiple versions in `.python-version`.

So, this PR adds a `make setup-pyenv` that automates that install idempotently.

Given that we may be switching CI soon, and that the py3 project is on pause right now, I'm not in favor of adding this to direnv for the time being.

Since we aren't enforcing pyenv in dev envs yet, `PYENV_VERSION=3.6.7` will just noop. **But pyenv users will have to run `make setup-pyenv` when this is merged, if they don't have that version.** I've added a nice check + prompt for that.

Note: you can't just specify a minor version, an explicit patch is needed. So existing pyenv users may need to install a new 3.
A good choice for this is 3.6.7 since it's the py3 version used by Travis for their Xenial environments.

### CODE REVIEW

**Comment by joshuarli** [comment]
Calling it now, lint job is gonna fail and I'll have to patch it.

File: Makefile [hunk 1]
```diff
@@ -60,14 +63,18 @@ setup-git-config:
 setup-git: ensure-venv setup-git-config
 	@echo "--> Installing git hooks"
 	cd .git/hooks && ln -sf ../../config/hooks/* ./
+	@# This 3.x.x version needs to be synced with the version in .python-version.
+	@PYENV_VERSION=3.6.7 python3 -c '' || (echo 'Please run `make setup-pyenv` to install the required Python 3 version.'; exit 1)
```
Comment (wedamija) [inline Makefile:67]: Is there any way for us to keep this in sync? I imagine this will break at some point.

Comment (joshuarli) [reply Makefile]: https://github.com/getsentry/sentry/pull/18906/commits/7e3b8b2ced4cdd47933d98a9d147d2e3de93056e

File: Makefile [hunk 2]
```diff
@@ -2,6 +2,9 @@ PIP := python -m pip --disable-pip-version-check
 WEBPACK := NODE_ENV=production ./bin/yarn webpack
 YARN := ./bin/yarn
 
+# Currently, this is only required to install black via pre-commit.
+REQUIRED_PY3_VERSION := $(shell awk 'FNR == 2' .python-version)
```
Comment (wedamija) [inline Makefile:6]: Wondering if it'd be safer to filter down to the first line that starts with `3.` instead? Either way this looks good

**Review by wedamija — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (joshuarli) [reply Makefile:6]: Eh, `.python-version` won't really be modified often.

**Pull request merged**